R/get_predicted.R

```diff
@@ -1225,7 +1222,27 @@ get_predicted.faMain <- function(x, data = NULL, ...) {
                                                  iter,
                                                  centrality_function = base::mean,
                                                  ...) {
-  predictions <- apply(iter, 1, centrality_function)
+
+  # outcome: ordinal/multinomial/multivariate produce a 3D array of predictions,
+  # which we stack in "long" format
+  if (length(dim(iter)) == 3) {
+    # 3rd dimension of the array is the response level. This stacks the draws into:
+    # Rows * Response ~ Draws
+    iter_stacked <- apply(iter, 1, c)
+    predictions <- data.frame(
+      # rows repeated for each response level
+      Row = rep(1:ncol(iter), times = dim(iter)[3]),
+      # response levels repeated for each row
+      Response = rep(dimnames(iter)[[3]], each = dim(iter)[2]),
+      Predicted = apply(iter_stacked, 1, centrality_function)
+    )
```

Comment:
Just to be safe, not sure if `Response` can be a character, should we add `stringsAsFactors = FALSE`?

Reply:
Good call. It's only been a 1.5 years, but I often forget that `stringsAsFactors` was once `TRUE` by default. Force pushed.